### PR TITLE
Change the name of repodata.json to pyodide-lock.json

### LINKED ISSUE #3788: Change `repodata.json` name to something else?
I am told by conda people that repodata.json is supposed to be the input to a solver. Our repodata.json is a lock file which is the output of a solver. This is a misleading name clash. We will need to spend a lot of time explaining to conda people that this file isn't what they expect to be. Can we rename it?

### PR DESCRIPTION
**Pull request by hoodmane**
Resolves #3788

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry

### CODE REVIEW

File: src/js/load-package.ts [hunk 1]
```diff
@@ -17,28 +17,28 @@ import { makeWarnOnce } from "./util";
...
+  if (!lockfile.packages) {
     throw new Error(
-      "Loaded repodata.json does not contain the expected key 'packages'.",
+      "Loaded pyodide lock file does not contain the expected key 'packages'.",
     );
   }
-  API.repodata_info = repodata.info;
-  API.repodata_packages = repodata.packages;
+  API.repodata_info = lockfile.info;
+  API.repodata_packages = lockfile.packages;
```
Comment (ryanking13) [inline src/js/load-package.ts:41]: Probably we should update the variable name in `micropip` and here together.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane. I think the name `pyodide-lock.json` is good.

File: src/js/load-package.ts (hunk 1 above)
Comment (hoodmane) [reply src/js/load-package.ts:41]: For compatibility reasons we could store it into both `repodata_xxx` and `lockfile_xxx` for a while.

**Review by rth — APPROVED** [review]
LGTM as well.

**Pull request merged**